### ISSUE TITLE
MetaBAT2. binning error.

### ISSUE THREAD

**Issue by zzjl20** [bxlab/metaWRAP#309]
I am running the metawrap binning with metaBAT2 only. The command is:
```
metawrap binning -t 32 --metabat2 --run-checkm -o /NGS/BINNING/ACHK -a /NGS/ASSEMBLE/ACHK/final_assembly.fasta /NGS/raw_data/ACHK_read_1.fastq /NGS/raw_data/ACHK_read_2.fastq
```
And the error happend when it comes to binning step:
```
------------------------------------------------------------------------------------------------------------------------
-----                                       Starting binning with metaBAT2...                                      -----
------------------------------------------------------------------------------------------------------------------------

MetaBAT 2 (v2.12.1) using minContig 1500, minCV 1.0, minCVSum 1.0, maxP 95%, minS 60, and maxEdges 200. 
/NGS/soft/miniconda2/bin/metawrap-modules/binning.sh: line 295: 18037 Segmentation fault      metabat2 -i ${out}/work_files/assembly.fa -a ${out}/work_files/metabat_depth.txt -o ${out}/metabat2_bins/bin -m $metabat_len -t $threads --unbinned

************************************************************************************************************************
*****                              Something went wrong with running MetaBAT2. Exiting                             *****
************************************************************************************************************************
```
Could you tell me how to deal with this problem? If you need more info please tell. Thank you very much.

**Comment by ursky** [comment]
Look into metabat2 errors and try running it in isolation until you get it to run. It could be your installation/environment, or the data itself. Good luck.

**Comment by zzjl20** [comment]
OK, I will give a try and report back.

**Comment by qikongwanli** [comment]
I got the same issues in metawrap=1.3.2. However, when I run metabat2=2.15 in another env, everythings were ok. So,  we should update the versions of metabat2=2.12 to metabat2=2.15. Think you @ursky

**Comment by famiji** [comment]
> I got the same issues in metawrap=1.3.2. However, when I run metabat2=2.15 in another env, everythings were ok. So, we should update the versions of metabat2=2.12 to metabat2=2.15. Think you @ursky

Could you please peovide more details about how to solve this problem? When I try 'mamba install metawrap2==2.15' in my metawrap env, it always fail and show"Could not solve for environment specs".

**Comment by qikongwanli** [comment]
> > I got the same issues in metawrap=1.3.2. However, when I run metabat2=2.15 in another env, everythings were ok. So, we should update the versions of metabat2=2.12 to metabat2=2.15. Think you @ursky
> 
> Could you please peovide more details about how to solve this problem? When I try 'mamba install metawrap2==2.15' in my metawrap env, it always fail and show"Could not solve for environment specs".

Yes, we cannot update metabat2=2.12 to metabat2=2.15 in metawrap environment. But I copied the file of metabat2=2.15 in new env, and directly overwrote the metabat2 files in the old metawrap environment.
Thank you.

**Comment by famiji** [comment]
Thanks for your reply, I also applied the way your mentioned above to solve
this problem and got the right result.
Thank you once again for providing the solution to this problem to me.

qikongwanli ***@***.***> 于 2023年8月23日周三 20:41写道：
